packages/compiler-core/src/transforms/vIf.ts

```diff
@@ -212,7 +212,10 @@ function createChildrenCodegenNode(
     const vnodeCall = (firstChild as ElementNode)
       .codegenNode as BlockCodegenNode
     // Change createVNode to createBlock.
-    if (vnodeCall.type === NodeTypes.VNODE_CALL) {
+    if (
+      vnodeCall.type === NodeTypes.VNODE_CALL &&
```

Comment:
Let's add a comment here: "component vnodes are always tracked and its children are compiled into slots so no need to make it a block"